mypy_django_plugin/transformers/managers.py

```diff
@@ -0,0 +1,153 @@
+from typing import Tuple, List
+
+from mypy.nodes import (
+    GDEF, Argument, FuncDef, MemberExpr, RefExpr, NameExpr, PlaceholderNode,
+    StrExpr, SymbolTableNode, TypeInfo, Var,
+)
+from mypy.plugin import ClassDefContext
+from mypy.types import AnyType, CallableType, Instance, TypeOfAny
+
+from mypy_django_plugin.lib import fullnames, helpers
+from mypy.plugins.common import add_method_to_class
+
+from mypy.types import Type as MypyType
+
+
+def build_unannotated_method_args(method_node: FuncDef) -> Tuple[List[Argument], MypyType]:
+    prepared_arguments = []
+    for argument in method_node.arguments[1:]:
+        argument.type_annotation = AnyType(TypeOfAny.unannotated)
+        prepared_arguments.append(argument)
+    return_type = AnyType(TypeOfAny.unannotated)
+    return prepared_arguments, return_type
+
+
+class ManagerCallback(helpers.DynamicClassFromMethodCallback):
+
+    def copy_method_to_another_class(
+            self,
+            ctx: ClassDefContext,
+            self_type: Instance,
+            new_method_name: str,
+            method_node: FuncDef) -> None:
+        if method_node.type is None:
+            if self.defer_till_next_iteration():
+                return
+
+            arguments, return_type = build_unannotated_method_args(method_node)
+            add_method_to_class(
+                    ctx.api,
+                    ctx.cls,
+                    new_method_name,
+                    args=arguments,
+                    return_type=return_type,
+                    self_type=self_type)
+            return
+
+        method_type = method_node.type
+        if not isinstance(method_type, CallableType):
+            if not self.semanal_api.final_iteration:
+                self.semanal_api.defer()
+            return
+
+        arguments = []
+        bound_return_type = self.semanal_api.anal_type(
+                method_type.ret_type,
+                allow_placeholder=True)
+
+        assert bound_return_type is not None
+
+        if isinstance(bound_return_type, PlaceholderNode):
+            return
+
+        for arg_name, arg_type, original_argument in zip(
+                method_type.arg_names[1:],
+                method_type.arg_types[1:],
+                method_node.arguments[1:]):
+            bound_arg_type = self.semanal_api.anal_type(arg_type, allow_placeholder=True)
+            if bound_arg_type is None and not self.semanal_api.final_iteration:
+                self.semanal_api.defer()
```

Comment:
Same thing as above, `defer_till_next_iteration` or/and exception raise, instead of just `return`.